### PR TITLE
Merges map templates and map loader

### LINKED ISSUE #23125: dmm_suite load_map() not initializing things loaded
Nothing loaded is initialized, this makes APCs , comms, and everything else not working.

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #23125

### CODE REVIEW

File: code/modules/mapping/reader.dm [hunk 1]
```diff
@@ -26,6 +26,12 @@ var/global/dmm_suite/preloader/_preloader = new
  *
  */
 /dmm_suite/load_map(dmm_file as file, x_offset as num, y_offset as num, z_offset as num, cropMap as num, measureOnly as num)
+	//How I wish for RAII
```
Comment (RemieRichards) [inline code/modules/mapping/reader.dm]: RAII?

**Comment by coiax** [comment]
HE'S TOUCHING MY THIIIIIIIIINGS

File: code/modules/mapping/reader.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/mapping/reader.dm]: C++ thing, it would basically eliminate the need for the wrapper proc

**Comment by AnturK** [comment]
Conflicts.

File: code/modules/mapping/reader.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/mapping/reader.dm]: RAII is a mistake

**Pull request merged**